File: Code/0/4/0412/0412.cpp
/******************************************************************************
 * 文件名: 0412
 * 作者: BOLUN XU
 * 创建日期: 2025
 * 版本: 1.0
 * 描述: Fizz Buzz。
 * 时间复杂度： O(N)
 * 空间复杂度： O(N)
 * 执行用时： 4 ms
 * 消耗内存： 11.61 mb
 ******************************************************************************/
#include <iostream>
#include <algorithm>
#include <vector>
#include <string>

using namespace std;

class Solution {
public:
    vector<string> fizzBuzz(int n) {
        vector<string> ans;
        for (int i = 1; i <= n; i++) {
          if (i % 3 == 0 && i % 5 == 0) {
            ans.push_back("FizzBuzz");
          }
          else if (i % 3 == 0) {
            ans.push_back("Fizz");
          }
          else if (i % 5 == 0) {
              ans.push_back("Buzz");
          }
          else {
            ans.push_back(to_string(i));
          }
        }
        return ans;
    }
};

File: Code/1000/0412/0412.py
#!/usr/bin/env python3
# -*- coding: utf-8 -*-

"""
文件名: PersonalLeetCode
作者: Bolun Xu
创建日期: 2025
版本: 1.0
描述: Fizz Buzz。
时间复杂度： O(N)
空间复杂度： O(N)
执行用时： 0 ms
消耗内存：18.02 mb
"""
from typing import List


class Solution:
    def fizzBuzz(self, n: int) -> List[str]:
        ans = []
        for i in range(1, n + 1):
            if i % 3 == 0 and i % 5 == 0:
                ans.append("FizzBuzz")
            elif i % 3 == 0:
                ans.append("Fizz")
            elif i % 5 == 0:
                ans.append("Buzz")
            else:
                ans.append(str(i))
        return ans


if __name__ == '__main__':
    solution = Solution()
    print(solution.fizzBuzz(15))
